qrscan/models.py

```diff
@@ -0,0 +1,123 @@
+import uuid
+from io import BytesIO
+from PIL import Image
+import qrcode
+import secrets
+
+from django.core.files import File
+from django.db import models
+
+from home.models import Allocation, Student
+
+
+def generate_secret_key():
+    return secrets.token_hex(32)
+
+
+class MessCard(models.Model):
+    id = models.UUIDField(
+        primary_key=True,
+        default=uuid.uuid4,
+        editable=False,
+        unique=True,
+        help_text="This contains the unique id of the mess card"
+    )
+    allocation = models.ForeignKey(
+        Allocation,
+        on_delete=models.CASCADE,
+        help_text="This contains the allocation details",
+        null=True,
+        blank=True
+    )
+    student = models.ForeignKey(
+        Student,
+        on_delete=models.CASCADE,
+        help_text="This contains the student details",
+        null=True,
+        blank=True
+    )
+    qr_code = models.ImageField(
+        upload_to='qr_codes/',
+        help_text="This contains the qr code image",
+        blank=True,
+        null=True
+    )
+
+    def __str__(self):
+        return f"{self.allocation.student_id} - {self.allocation.email.email}"
+
+    def generate_qr_code(self):
+        data = str(self.id)
+
+        qr = qrcode.QRCode(
+            version=1,
+            error_correction=qrcode.constants.ERROR_CORRECT_H,
+            box_size=10,
+            border=4,
+        )
+        qr.add_data(data)
+        qr.make(fit=True)
+
+        img = qr.make_image(fill="black", back_color="white").convert('RGB')
+
+        try:
+            logo = Image.open('./static/images/iiti_bw.png')
+            logo = logo.resize((150, 150))
+            pos = ((img.size[0] - logo.size[0]) // 2, (img.size[1] - logo.size[1]) // 2)
+            img.paste(logo, pos)
```

Comment:
why are we doing this?

Reply:
It adds iiti image to the center of the qr to look more professional